src/TypeSpec.Extension/Emitter.Csharp/package.json

```diff
@@ -40,7 +40,7 @@
     },
     "peerDependencies": {
         "@azure-tools/typespec-azure-core": ">=0.36.0 <1.0.0",
-        "@azure-tools/typespec-client-generator-core": ">=0.36.0 <1.0.0",
+        "@azure-tools/typespec-client-generator-core": ">=0.36.0 <1.0.0 || 0.40.0-dev.14",
```

Comment:
Why not just >= 0.40.0-dev.14? We've introduced dev versions anyway.

Reply:
Because it is just a workaround. We cannot generate lock file without this change. But we will remove this after we upgrade to a release version.